src/transformers/models/vision_encoder_decoder/modeling_vision_encoder_decoder.py

```diff
@@ -653,11 +653,19 @@ def prepare_inputs_for_generation(
     ):
         decoder_inputs = self.decoder.prepare_inputs_for_generation(input_ids, past_key_values=past_key_values)
         decoder_attention_mask = decoder_inputs["attention_mask"] if "attention_mask" in decoder_inputs else None
+        if kwargs.get("decoder_inputs_embeds") is not None:
+            decoder_inputs["input_ids"] = None
+            decoder_inputs_embeds = self.decoder.prepare_inputs_for_generation(kwargs.get("decoder_inputs_embeds"), past_key_values=past_key_values)
+            decoder_inputs_embeds = decoder_inputs_embeds["input_ids"]
```

Comment:
This does not follow the reference implementation, see [here](https://github.com/huggingface/transformers/pull/21575) for an example